pkg/meta/redis/service/redis.go

```diff
@@ -223,60 +236,49 @@ func (c *client) Set(key, val string) error {
 	return vk.Err()
 }
 
-func (c *client) SetMultiple(kvs map[string]string) error {
-	kvl := make([]interface{}, 0, len(kvs)*2)
-	vkl := make([]interface{}, 0, len(kvs)*2)
+func (c *client) SetMultiple(kvs map[string]string) (err error) {
+	pipe := c.db.TxPipeline()
+	ress := make([]redis.StatusCmd, 0, len(kvs)*2)
 	for k, v := range kvs {
 		if len(k) == 0 || len(v) == 0 {
 			continue
 		}
-		kvl = append(kvl, k, v)
-		vkl = append(kvl, v, k)
+		ress = append(ress, pipe.Set(c.appendPrefix(c.kvPrefix, k), v, 0), pipe.Set(c.appendPrefix(c.vkPrefix, v), k, 0))
 	}
-	pipe := c.db.TxPipeline()
-	pipe.Select(c.kvIndex)
-	kv := pipe.MSet(kvl...)
-	pipe.Select(c.vkIndex)
-	vk := pipe.MSet(vkl...)
-	if _, err := pipe.Exec(); err != nil {
+	if _, err = pipe.Exec(); err != nil {
 		return err
 	}
-	if err := kv.Err(); err != nil {
-		return err
+	var errs error
+	for _, res := range ress {
+		errs = errors.Wrap(errs, res.Err().Error())
```

Comment:
could you please check res.Err is nil or not here, also errors.Wrap checks error and message, but it is depends on library